//syntax = "proto3";
//option cc_generic_services =true;
//package rpcService;
//
//message AppendEntry {
//     int64 term  = 1;
//     int64 leaderId  = 2;
//     int32 prevLogIndex = 3;
//     int32 prevLogTerm =4;
//     message LogEntry{
//          int64 index = 1;
//          int64 term = 2;
//          string commandName =3 ;
//          bytes command = 4;
//     }
//     repeated LogEntry LogEntries = 5;
//}
//
//message response {
//     int64 Term  = 1;
//     int64 Index = 2;
//     int64 CommitIndex =3;
//     bool Success  = 4;
//}
//
//service EchoService {
//     rpc Foo(request) returns(response);
//}


syntax = "proto3";
option cc_generic_services =true;
package rpcService;



message AppendEntriesRequest {
     int64 term  = 1;
     int64 leaderId  = 2;
     int32 prevLogIndex = 3;
     int32 prevLogTerm =4;
     message LogEntry{
          int64 index = 1;
          int64 term = 2;
          string commandName =3 ;
          bytes command = 4;
     }
     repeated LogEntry LogEntries = 5;
}

message AppendEntriesResponse {
     int64 Term  = 1;
     int64 Index = 2;
     int64 CommitIndex =3;
     bool Success  = 4;
}

service AppendEntriesService {
     rpc AppendEntries(AppendEntriesRequest) returns(AppendEntriesResponse);
}